src/main/resources/org/jenkinsci/plugins/mesos/MesosCloud/config.jelly

```diff
@@ -19,12 +19,8 @@
         <f:textbox field="slavesUser" default=""/>
     </f:entry>
 
-    <f:entry title="${%Framework Principal}" field="principal">
-      <f:textbox field="principal" default="jenkins" />
-    </f:entry>
-
-    <f:entry title="${%Framework Secret}" field="secret" description="${%Specify secret to enable framework authentication}">
-      <f:textbox field="secret"/>
+    <f:entry field="credentialsId" title="${%Framework credentials}">
```

Comment:
I retain the fields as a matter of backward compatibility, otherwise it is much more complicated code-wise to migrate the existing data.